pkg/controllers/metrics/termination/controller.go

```diff
@@ -0,0 +1,139 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package termination
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/logging"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/prometheus/client_golang/prometheus"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/metrics"
+	"github.com/aws/karpenter/pkg/utils/functional"
+)
+
+const (
+	controllerName = "terminationmetrics"
+	provisioner    = "provisioner"
+)
+
+var (
+	terminationProvisionerSummaryVec = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Namespace:  "karpenter",
+			Subsystem:  "nodes",
+			Name:       "termination_provisioner_time_seconds",
+			Help:       "The time taken between a node's deletion request and its termination, labeled by provisioner",
```

Comment:
Thoughts on labeling this metric with all well known labels instead of just provisioner? e.g. How long is it taking large instances to scale down compared to small ones? Does it vary by zone? etc.

Reply:
The problem is it is not possible to aggregate summary metrics. Because of this, for summaries to be useful, we would have a combinitorial explosion of metric families. @njtran and I decided that provisioner was likely the most useful label to be able to filter on so I included an additional metric family to allow filtering on provisioners. A more detailed explanation is in the updated doc's appendix.